content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -275,33 +275,79 @@ node {
 }
 ----
 
-=== Defining steps
 
+=== Defining global variables
+
+Internally, scripts in the `vars` directory are instantiated as a singleton
+on-demand, when used first.  So it is possible to define more methods,
+properties on a single file that interact with each other:
+
+[source,groovy]
+----
+// vars/acme.groovy
+def setFoo(v) {
+    this.foo = v;
```

Comment:
[Verified](https://gist.github.com/rtyler/e32169ece1d2f51c4b9fe5c7d2883474) that this does not result in stack overflows